### PR TITLE
prova

### PR DESCRIPTION
**Pull request by calvinmetcalf**
hopefully testing on firefox in travis

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
this is tricky because I'm pretty sure firefox is getting the script not responding box which is impossible to click

**Comment by calvinmetcalf** [comment]
ok got it working by asyncifying the pbkdf2 and omitting much of the diffie-hellman,

File: test/dh.js [hunk 1]
```diff
@@ -22,26 +22,28 @@ test('diffie-hellman mod groups', function (t) {
...
-      dh1.generateKeys();
-      dh2.generateKeys();
-      t.equals(p1, p2, 'equal primes');
-      var pubk1 = dh1.getPublicKey();
-      var pubk2 = dh2.getPublicKey();
-      t.notEquals(pubk1, pubk2, 'diff public keys');
-      var pub1 = dh1.computeSecret(pubk2).toString('hex');
-      var pub2 = dh2.computeSecret(dh1.getPublicKey()).toString('hex');
-      t.equals(pub1, pub2, 'equal secrets');
+if (!process.browser) {
```
Comment (dcousens) [inline test/dh.js]: can you elaborate on this?  I'm not sure I follow.

File: test/pbkdf2.js [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
 var pbkdf2Sync = require('../').pbkdf2Sync
 
 var vectors = require('hash-test-vectors/pbkdf2')
 
 tape('pbkdf2', function (t) {
+	t.plan(vectors.length);
   vectors.forEach(function (input) {
-    //skip inputs that will take way too long
-    if(input.iterations > 10000) return
+  	setTimeout(function () {
```
Comment (dcousens) [inline test/pbkdf2.js]: Why?  I know you mentioned it, but this seems like a hack.

Comment (calvinmetcalf) [inline test/pbkdf2.js]: its to prevent the script from locking the page up to long,

File: test/dh.js (hunk 1 above)
Comment (calvinmetcalf) [inline test/dh.js]: these tests take a LONG time so on a very very slow travis browser they time out/lock the browser

**Comment by calvinmetcalf** [comment]
ok I removed the prepublish script and replaced it with one called publish which runs npm ls, followed by npm run test-node and then runs npm publish, this is because republish runs after npm install but if the tests fail there then instead of just failing travis it causes npm install to be fun again up to 3 times.

This changes the tests in 2 ways

- the pbkdf2 tests are changed to be async, this is to prevent locking the browser which ( I believe) causes a script not responding dialog to show up in firefox which we can't then click through so the test hangs
- the diffie-hellman tests are truncated to only do a few of the comparisons and not generate any primes.   Again this is do to them taking a while and locking up the browser, I might be able to do something overly complicated with a web worker but that seems like a phase 2 to me.

Before merging I can squash

**Comment by dominictarr** [comment]
:+1:

**Comment by calvinmetcalf** [comment]
ok rebased and squashed going to make sure it's still green

File: test/dh.js (hunk 1 above)
Comment (dcousens) [inline test/dh.js]: I feel like this is just a travis artifact, and shouldn't be omitted from general browser testing.
How would I test this code in a browser manually without having to omit these statements?

**Comment by dcousens** [comment]
Before merge, can we ensure we aren't just disabling tests simply because travis times out?

**Comment by calvinmetcalf** [comment]
Ok won't merge yet

**Comment by calvinmetcalf** [comment]
trying to get the diffie-hellmen tests to work in a web worker leading to some circuler deps issues, looks like diffie-hellmen will be the next one to break out

**Comment by calvinmetcalf** [comment]
closing as it's bit rotted

**Pull request closed without merging**